### PR TITLE
New package: hledger-1.2

### PR DESCRIPTION
**Pull request by ibabushkin**
`hledger` was requested in #5706. This currently splits stuff in subpackages, I still need assurance from travis to see whether some fixes are needed.

### CODE REVIEW

**Comment by ibabushkin** [comment]
To kick off the discussion. `stack` doesn't build on musl. Any solutions besides building only for... well, non-musl (which is essentially only `i686` and `x86_64`).

File: srcpkgs/hledger/template [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Template file for 'hledger'
+pkgname=hledger
+version=1.2
+revision=1
+build_style=haskell-stack
+nocross=yes
```
Comment (Vaelatern) [inline srcpkgs/hledger/template]: Is there any real reason to keep this?

File: srcpkgs/hledger/template [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+homepage="http://hledger.org/"
+hostmakedepends="ghc stack"
+makedepends="zlib-devel ncurses-devel"
+wrksrc="${pkgname}-${pkgname}-${version}"
+stackage="lts-8.6"
+distfiles="https://github.com/simonmichael/${pkgname}/archive/${pkgname}-${version}.tar.gz"
+checksum="bd00115272feada9e388b1aafe6f105369f0d0f0286f65b4ed35fb4ffcd3ad54"
+nopie=yes  # network-2.6.3.1 build fails with a relocation
+
+subpackages="hledger-ui hledger-web hledger-api"
```
Comment (Vaelatern) [inline srcpkgs/hledger/template]: This variable is redundant.

File: srcpkgs/hledger/template [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Template file for 'hledger'
+pkgname=hledger
+version=1.2
+revision=1
+build_style=haskell-stack
+nocross=yes
+short_desc="Hledger is an accounting program for money, time, or other commodities"
```
Comment (Vaelatern) [inline srcpkgs/hledger/template]: `short_desc="Simple, precise, plain text accounting"`

File: srcpkgs/hledger/template (hunk 1 above)
Comment (Vaelatern) [inline srcpkgs/hledger/template]: Nvm, yes.

File: srcpkgs/hledger/template (hunk 2 above)
Comment (ibabushkin) [inline srcpkgs/hledger/template]: Yeah, the order of subpackages shouldn't be important.

File: srcpkgs/hledger/template (hunk 1 above)
Comment (ibabushkin) [inline srcpkgs/hledger/template]: what bit? the build style or the nocross setting?

**Comment by Vaelatern** [comment]
I'm currently testing my modifications and will push soon

**Comment by ibabushkin** [comment]
Yeah, I'm waiting for travis to finish.

**Comment by ibabushkin** [comment]
Turns out, waiting was useless.

**Comment by chneukirchen** [comment]
We don't have ghc on musl, so...

**Comment by ibabushkin** [comment]
That was to be expected :P I guess I'll add the arch restriction then, so we only have non-musl `i686` and `x86_64`. What should we do about the travis timeout?

**Comment by Vaelatern** [comment]
```
WARNING: Installation path /destdir/hledger-1.2/usr/bin not found on the PATH environment variable
=> hledger-1.2_1: running post_install ...
=> ERROR: hledger-1.2_1: vinstall: cannot find 'hledger/doc/other/*.5'...
=> ERROR: hledger-1.2_1: failed to run post_install() at line 12.
```
on x86_64

**Comment by Vaelatern** [comment]
Don't add an arch restriction.

**Comment by Vaelatern** [comment]
I think I fixed that problem I just shared. Re-running.

**Comment by ibabushkin** [comment]
Nice! How exactly? I'm a bit baffled about why the paths don't match the directory layout of the tarball.

**Comment by Vaelatern** [comment]
vman does not take wildcards, so instead I'm looping over the relevant files and vmaning each one

**Comment by ibabushkin** [comment]
ah, stupid me, that makes sense. I suppose the warning regarding paths is irrelevant due to the chroot environment?

**Comment by Vaelatern** [comment]
Build time (MM:SS): 39:59 with 24 cores at it.
But it works for me after simple testing, and the packages seem to have what they need, so for now, merging!

**Comment by ibabushkin** [comment]
Alright, let's see how this works out! I already thought about how we could reduce the perverse build time, so stay tuned :)

**Pull request closed without merging**